burn-core/src/grad_clipper/base.rs

```diff
@@ -0,0 +1,69 @@
+use burn_tensor::{backend::Backend};
+use crate::tensor::Tensor;
+/// Gradient Clipper provides a way to mitigate the exploding gradients
+/// problem by clipping every component of the gradient to a value during 
+/// backpropagation.
+pub struct GradientClipper {
+    clip_value: Option<f32>,
+    clip_norm: Option<f32>,
+}
+
+impl GradientClipper {
+    pub fn by_value(threshold: f32) -> Self {
+        Self {
+            clip_value: Some(threshold),
+            clip_norm: None,
+        }
+    }
+
+    pub fn by_norm(threshold: f32) -> Self {
+        Self {
+            clip_value: None,
+            clip_norm: Some(threshold),
+        }
+    }
+
+    pub fn clip_gradient<B: Backend, const D: usize>(&self, grad: Tensor<B, D>) -> Tensor<B, D> {
+        match (self.clip_value, self.clip_norm) {
+            (Some(clip_value), None) => self.clip_by_value(grad, clip_value),
+            (None, Some(clip_norm)) => self.clip_by_norm(grad, clip_norm),
+            _ => panic!("Gradient Clipper must have one (and only one) clip_value or clip_norm."),
+        }
+    }
+
+    fn clip_by_value<B: Backend, const D: usize>(&self, grad: Tensor<B, D>, threshold: f32) -> Tensor<B, D> 
+    where
+        B::FloatElem: core::cmp::PartialOrd<f32> + From<f32>, // this doesn't feel right
```

Comment:
@nathanielsimard I know that the gradients coming in are going to be made up of f32's, but it seems like this is not a very burn-ish way of constraining the inputs. Do you have any recommendations?

Reply:
You have the type `B::FloatElem` in the backend. You can also convert any numeric types into `B::FloatElem` using `elem()` function. You have to import `ElementConversion` trait I think. I would probably keep f32 in the struct and convert is when clip is called.